src/NuGet.Core/NuGet.Packaging.Core/IPackageCoreWriter.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Packaging.Core
+{
+    /// <summary>
+    /// Allows package modifications.
+    /// </summary>
+    public interface IPackageCoreWriter
+    {
+        /// <summary>
+        /// Remove if the path exists.
+        /// </summary>
+        /// <param name="path">Relative file path in package.</param>
+        /// <returns>True if the file existed before the remove.</returns>
+        /// <param name="token">CancellationToken</param>
+        Task<bool> RemoveAsync(string path, CancellationToken token);
```

Comment:
Will this remove empty directories on file deletes?

Reply:
I would expect so, for zip files I'm not sure that is an issue since directories aren't their own entries unless they are empty from the start